### ISSUE TITLE
Additional features

### ISSUE THREAD

**Issue by jimfleming** [fomorians/pyoneer#3]
I went through the #api channel and found our wishlists. Some of these may already be implemented in this repo and likely implemented _somewhere_ in our codes.

- [x] Numeric stability utilities to avoid dividing by zero:

    ```
    tf.where(outputs_variances < 1e-6,
        tf.ones_like(outputs_variances) * 1e-6,
        outputs_variances)
    ```

- [x] Functions and Keras layers for normalization by mean/std and min-max scaling.
- [ ] Keras layers/models for correlated/uncorrelated noise, wrapping distributions.
- [ ] Concrete dropout
- [ ] Stateless dropout for Bernoulli/Gaussian/Concrete for CRN/PEGASUS
- [x] Gumbel softmax (tensorflow_probability might already support this)
- [ ] Subsampling for sequences: deterministic and overlapping (good for image channels, text, and rollouts)
- [x] General padding utilities
- [x] Swish...
- [x] `policy_ratio_loss`
- [x] TF-compatible `is_close`
- [x] Streaming normalizer (for rewards and such)
- [x] Keras RNN class
- [x] `compute_returns`
- [x] `compute_advantages`
- [x] `clip_gradients`
- [x] `copy_variables` (.assign between two lists of variables, optionally weighted for soft-copying)
- [ ] Utilities for temporal masking of convolutions, e.g. WaveNet, etc.
